### PR TITLE
v20 beta 2: AbiProvider, Export Changes, and Clean up

### PR DESCRIPTION
**Pull request by c0d3ster**
**Breaking Changes**

Changes include:
- Inclusion of optional AbiProvider in Api constructor (allows passing of local abi's as opposed to retrieving them from the chain)
- Moved tests and setup for tests into their own directory
- Remove default export in index.js to enable tree shaking in the future
- Pulled interfaces into their own files and exported them separately
- new Rpc.JsonRpc() -> new JsonRpc()
- new SignatureProvider() -> new JsSignatureProvider() (clarifies danger of bringing keys into webpage context)
- Export RpcError
- Changed BinaryAbi interface account_name field to camel case
- Updated README.md with new syntax and additional information
- set alwaysStrict flag in TSC
- Overall code cleanup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,34 +16,45 @@ Documentation can be found [here](https://eosio.github.io/eosjs)
```
Comment (TaraTritt) [inline README.md]: Under the installation header a few lines above, there are only NodeJS installation instructions, you might want to either remove that header (### NodeJS) or add that it also works for the browser with projects using npm

File: README.md [hunk 2]
```diff
@@ -16,34 +16,45 @@ Documentation can be found [here](https://eosio.github.io/eosjs)
...
+### Browser 
+
+Importing using ES6 module syntax in the browser is supported if you have a transpiler, such as Babel.
+```js
+import { Api, JsonRpc, RpcError, JsSignatureProvider } from 'eosjs';
+```
+
 ### NodeJS
 
+Importing using commonJS syntax is supported by node out of the box.
```
Comment (TaraTritt) [inline README.md]: Change `node` to  `NodeJS` as you reference it that way everywhere else

Comment (TaraTritt) [reply README.md]: Probably should capitalize commonJS as well - CommonJS

File: README.md [hunk 3]
```diff
@@ -16,34 +16,45 @@ Documentation can be found [here](https://eosio.github.io/eosjs)
...
 ```js
-const eosjs = require('eosjs');
+const { Api, JsonRpc, RpcError, JsSignatureProvider } = require('eosjs');
 const fetch = require('node-fetch');                            // node only; not needed in browsers
 const { TextDecoder, TextEncoder } = require('text-encoding');  // node, IE11 and IE Edge Browsers
 ```
 
 ### SignatureProvider
```
Comment (TaraTritt) [inline README.md]: Should this header be `JsSignatureProvider` now and also referenced as `JsSignatureProvider` right below too?

File: README.md [hunk 4]
```diff
@@ -16,34 +16,45 @@ Documentation can be found [here](https://eosio.github.io/eosjs)
...
-Open a connection to JSON-RPC, include `fetch` when on NodeJS
+Open a connection to JSON-RPC, include `fetch` when on NodeJS.
 ```js
-const rpc = new eosjs.Rpc.JsonRpc('http://127.0.0.1:8000', { fetch });
+const rpc = new JsonRpc('http://127.0.0.1:8000', { fetch });
 ```
 
 ### API Constructor
 
 Include textDecoder and textEncoder when using in browser.
```
Comment (TaraTritt) [inline README.md]: Include textDecoder and textEncoder when using in `the` browser.

File: README.md [hunk 5]
```diff
@@ -75,15 +86,15 @@ const api = new eosjs.Api({ rpc, signatureProvider, textDecoder: new TextDecoder
 
 ### Error handling
 
-use `eosjs_jsonrpc.RpcError` for handling JSON-RPC Errors
+use `RpcError` for handling RPC Errors
```
Comment (TaraTritt) [inline README.md]: `Use` capitalized

Comment (TaraTritt) [reply README.md]: I think i preferred the JSON-RPC Errors  instead of just RPC from the older line, since it's the json rpc class

Comment (c0d3ster) [reply README.md]: We may implement other RPC formats in the future and the error will work for all formats, so I left this non-specific

**Pull request merged**